cogeo_mosaic/create.py

```diff
@@ -123,22 +173,11 @@ def create_mosaic(
             continue
 
         intersections_geoms = [dataset_geoms[idx] for idx in intersections_idx]
-        intersections = [results[idx] for idx in intersections_idx]
-
-        dataset = [
-            {"path": f["properties"]["path"], "geometry": geom}
-            for (f, geom) in zip(intersections, intersections_geoms)
-        ]
-
-        if minimum_tile_cover is not None or tile_cover_sort:
-            dataset = _filter_and_sort(
-                tile_geom,
-                dataset,
-                minimum_cover=minimum_tile_cover,
-                sort_cover=tile_cover_sort,
-            )
+        intersections = [features[idx] for idx in intersections_idx]
+
+        dataset = asset_filter(tile, intersections, intersections_geoms, **kwargs)
```

Comment:
@vincentsarago aka here, this is a function that takes in 

- `tile` (`mercantile.Tile`)
- `dataset` (List[Dict])
- `geoms` (pygeos polygons)
- `kwargs` any other parameters passed to the filter function.

This means that a user can easily override this function